hudi-common/src/main/java/org/apache/hudi/avro/AvroSchemaUtils.java

```diff
@@ -302,9 +306,17 @@ private static Schema appendFieldsToSchemaBase(Schema schema, List<Schema.Field>
    * @return schema with fields from fields, and metadata from schema
    */
   public static Schema createNewSchemaFromFieldsWithReference(Schema schema, List<Schema.Field> fields) {
+    if (schema == null) {
+      throw new IllegalArgumentException("Schema must not be null");
+    }
     Schema newSchema = Schema.createRecord(schema.getName(), schema.getDoc(), schema.getNamespace(), schema.isError());
-    for (Map.Entry<String, Object> prop : schema.getObjectProps().entrySet()) {
-      newSchema.addProp(prop.getKey(), prop.getValue());
+    Map<String, Object> schemaProps = schema.getObjectProps();
+    if (schemaProps != null) {
+      for (Map.Entry<String, Object> prop : schemaProps.entrySet()) {
+        newSchema.addProp(prop.getKey(), prop.getValue());
+      }
+    } else {
+      LOG.warn("Schema.getObjectProps() returned null for schema: {}", schema);
```

Comment:
do we even need to log a warn ?

Reply:
Schema has fields but no props. Throwing error would result in the same failure. The props were copied in https://github.com/apache/hudi/commit/eb20273cf662158f7872004c5460748b9695a240. We need to understand what's the side-effect of not having props. I think schema fields should be sufficient. @jonvex can help clarify this point.

> do we even need to log a warn ?

Logging a warn for debugging purpose.

Reply:
Yeah, I found the doc mentioned the attributes: https://avro.apache.org/docs/1.11.1/specification/#schema-declaration, it should not affect the serialization/deserialization, but it doesn't say how it is affected.